tests/unittest/compiler/test_fuse_split_cat.py

```diff
@@ -0,0 +1,262 @@
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+import unittest
+
+import torch
+from aitemplate.compiler import compile_model, ops
+
+from aitemplate.compiler.base import Tensor
+from aitemplate.compiler.ops.common.epilogue import FuncEnum
+from aitemplate.compiler.public import IntImm
+
+from aitemplate.testing import detect_target
+from aitemplate.testing.test_utils import get_random_torch_tensor, graph_has_op
+
+
+class FuseSplitCatTestCase(unittest.TestCase):
+    def _test_fuse_split_cat_rearrange(self, M, N, split, remove_split=True):
+        dtype = "float16"
+        M = IntImm(M)
+        N = IntImm(N)
+
+        input_1 = Tensor(
+            shape=[M, N],
+            name="input_1",
+            is_input=True,
+        )
+        split_2 = ops.split()(input_1, split, 0)
+        concatenate_3 = ops.concatenate()(split_2[::-1], 0)
+
+        # Set outputs
+        concatenate_3._attrs["name"] = "output_0"
+        concatenate_3._attrs["is_output"] = True
+        # Compile
+        model = compile_model(
+            concatenate_3, detect_target(), "./tmp", self._testMethodName
+        )
+        # Check that split was removed
+        self.assertEqual(
+            graph_has_op(model.debug_sorted_graph, "split"), not remove_split
+        )
+        # Run
+        input_1 = get_random_torch_tensor((M.value(), N.value()), dtype=dtype)
+        # Compare
+        split_pt = torch.split(input_1, split, 0)
+        y_pt = torch.cat(
+            [split_pt[1], split_pt[0]],
+            0,
+        )
+        y_ait = torch.empty_like(y_pt)
+        model.run_with_tensors(
+            {"input_1": input_1},
+            [y_ait],
+        )
+        torch.testing.assert_close(y_ait, y_pt, atol=0, rtol=0)
+
+    def test_fuse_split_cat_even(self):
+        self._test_fuse_split_cat_rearrange(
+            512, 512, split=[256, 256], remove_split=True
+        )
+
+    def test_fuse_split_cat_odd(self):
+        self._test_fuse_split_cat_rearrange(
+            512, 512, split=[139, 373], remove_split=True
+        )
+
+    def test_fuse_split_cat_reuse(self):
+        """Use a split output twice in the concatenate op."""
+        dtype = "float16"
+        M = IntImm(512)
+        N = IntImm(512)
+
+        input_1 = Tensor(
+            shape=[M, N],
+            name="input_1",
+            is_input=True,
+        )
+        split_2 = ops.split()(input_1, int(M.value() / 2), 0)
+        concatenate_3 = ops.concatenate()([split_2[1], split_2[0], split_2[1]], 0)
```

Comment:
This comment is related to the ```len(output.dst_ops()) == 1``` part that I put above. Could we add another test that similar to this but it has something like:

```
concatenate_3 = ops.concatenate()([split_2[1], split_2[0], split_2[1]], 0)
add_4 = ops.elementwise(FuncEnum.ADD)(split_2[1], another_tensor)
add_4._attrs["is_output"] = True
```